clients/src/test/java/org/apache/kafka/clients/producer/internals/ProducerBatchTest.java

```diff
@@ -65,7 +64,6 @@ public void testChecksumNullForMagicV2() {
         ProducerBatch batch = new ProducerBatch(new TopicPartition("topic", 1), memoryRecordsBuilder, now);
         FutureRecordMetadata future = batch.tryAppend(now, null, new byte[10], Record.EMPTY_HEADERS, null, now);
         assertNotNull(future);
-        assertNull(future.checksumOrNull());
```

Comment:
this test case (`testChecksumNullForMagicV2`) is meaningless after we remove  `assertNull(future.checksumOrNull())`. It seems to me it is ok to remove whole test case.